internal/app/singularity/cache_list_linux.go

```diff
@@ -16,6 +16,9 @@ import (
 	"github.com/sylabs/singularity/internal/pkg/sylog"
 )
 
+// findSize will take a int64 (like: '4781321234') and convert it to human readable output
+// in Kb, Mb, and Gb. output will be a string.
+// TODO: I'm sure theres a default 'human readable' function
 func findSize(size int64) (string, error) {
 	var sizeF float64
 	if size <= 1000000 {
```

Comment:
I know this is existing code and not what you are changing, but ...

The code cannot produce an error, as all possible values are covered (anything to 1000000; 1000001 to 1000000000; 1000000001 and up), so returning an error does not serve a useful purpose.

Note that because of the way the ranges are checked, findSize(1000000) will return "1000 kB" when it's likely that "1 MB" is expected. Same thing with findSize(1000000000) which returns "1000 MB" instead of "1 GB".

To help with the readability of this code, maybe this could be written like this:

```
var factor float64
var unit string
switch {
case size < 1e6:
    factor = 1e3
    unit = "kB"
case size < 1e9:
    factor = 1e6
    unit = "MB"
case size < 1e12:
    factor = 1e9
    unit = "GB"
default:
    factor = 1e12
    unit = "TB"
}

return fmt.Sprintf("%.2f %s, float(size) / factor, unit)
```